instant-xml/tests/forward-enum.rs

```diff
@@ -9,20 +11,50 @@ enum Foo {
     Baz(Baz),
 }
 
-#[derive(Debug, Eq, FromXml, PartialEq, ToXml)]
+#[derive(Clone, Debug, Eq, FromXml, PartialEq, ToXml)]
 struct Bar {
     bar: u8,
 }
 
-#[derive(Debug, Eq, FromXml, PartialEq, ToXml)]
+#[derive(Clone, Debug, Eq, FromXml, PartialEq, ToXml)]
 struct Baz {
     baz: String,
 }
 
+#[derive(Debug, PartialEq, ToXml, FromXml)]
+#[xml(forward)]
+enum FooCow<'a> {
+    Bar(Cow<'a, [BarBorrowed<'a>]>),
+    Baz(Cow<'a, [BazBorrowed<'a>]>),
+}
+
+#[derive(Clone, Debug, Eq, FromXml, PartialEq, ToXml)]
+#[xml(rename = "Bar")]
+struct BarBorrowed<'a> {
+    bar: Cow<'a, str>,
+}
+
+#[derive(Clone, Debug, Eq, FromXml, PartialEq, ToXml)]
+#[xml(rename = "Baz")]
+struct BazBorrowed<'a> {
+    baz: Cow<'a, str>,
+}
```

Comment:
Nit: let's move these below the `wrapped_enum()` test.

Why do we need all the `Clone` derives? Do we actually need them?

Reply:
> Nit: let's move these below the wrapped_enum() test.

Sure, fixed :) Any reason why you prefer to reduce the number of #[test]? Normally more inclined to do the opposite.

> Why do we need all the Clone derives? Do we actually need them?

Yes, I believe it's because of this.

https://github.com/djc/instant-xml/blob/0f641fdc1afaa4bd519741a31910b5e624e4e5c8/instant-xml/src/impls.rs#L330-L336

Other derives weren't needed, made sure to remove them.

Reply:
> > Nit: let's move these below the wrapped_enum() test.
> 
> Sure, fixed :) Any reason why you prefer to reduce the number of #[test]? Normally more inclined to do the opposite.

Sorry, I didn't mean "into" but "below", want to keep the types close to the test they belong with, so <types for test 1> <test 1> <types for test 2> <test 2> rather than <types for test 1> <types for test 2> <test 1> <test 2>.

> 
> > Why do we need all the Clone derives? Do we actually need them?
> 
> Yes, I believe it's because of this.
> 
> https://github.com/djc/instant-xml/blob/0f641fdc1afaa4bd519741a31910b5e624e4e5c8/instant-xml/src/impls.rs#L330-L336
> 
> Other derives weren't needed, made sure to remove them.

Okay, thanks.

Reply:
Ok gotchu. Would I be taking it too far if I place the corresponding struct definitions inside the corresponding tests?